### DISCUSSION TITLE
HTMLChart lines & bars colors

### DISCUSSION THREAD

**Discussion by amanfromfrance** [EvotecIT/PSWriteHTML#362] [Q&A] [answered]
Hello,

I am dynamically generating a report page that contains a "per customer" graph each week using the **HTMLChart** feature.
I used examples in the repository to build my template which is the following:

```
New-HTMLSection -Invisible {

	New-HTMLChart -Title $customer -TitleAlignment center {
		#New-ChartMarker -Size 30 -Color red -Shape square -StrokeColors yellow
		New-ChartAxisX -Name $dates
		New-ChartAxisY -TitleText 'Total Number of Connections' -Show
		New-ChartAxisY -TitleText 'Total Number of XXX Used' -Show -Opposite
		#New-ChartDataLabel -Enabled -DataLabelsColor Green, Yellow, Blue

		New-ChartLine -Name 'Total Number of Connections' -Value $numberOfConnections -Curve smooth -Width 2 -Dash 12 -Cap round -Color "#000000"
		New-ChartLine -Name 'Total Number of XXX Used' -Value $numberOfVmrs -Color "#8f8f8f"

		New-ChartBar -Name 'Total Number of Conferences' -Value $numberOfConferences
		#New-ChartBar -Name 'Test very long' -Value $DataTable.Resolved
	}
}
```

The generated graphs look like this:

![2023-01-24_090425](https://user-images.githubusercontent.com/58064665/214240494-c9e2bb55-7904-4fd2-88cb-d4680df50059.png)

As you can see I was able to change the **ChartLines** color to fit my requirements. However I am unable to set:

- The **ChartBar** color,
- The Legend and labels colors.

Do you have any idea how I can do that?

I already tried to play with the **ChartDataLabel** item without any effect...

Thanks for your help!

Julien

**Comment by PrzemyslawKlys** [comment, marked as answer]
Typically New-ChartLegend should work, but it doesn't seem to work. I guess I've not done extensive testing using mixed charts. Must be a bug to fix

**Reply by amanfromfrance** [reply]
Thanks for your quick feedback! Is that something that you plan to solve?

**Reply by PrzemyslawKlys** [reply]
Everything that is a bug should be solved. Once I get some time, I will fix everything. Usually, things like this are around 2 hours of play time.

**Reply by amanfromfrance** [reply]
Well noted, thanks again for your time!

**Reply by amanfromfrance** [reply]
In the meantime, I found a workaround to get the expected result: after the report generation, by modifying the **_new ApexCharts_** options of each graph in the HTML file I can set the correct colors:

![image](https://user-images.githubusercontent.com/58064665/214795714-85331fb1-c968-42fc-89c3-ff44e04d4efe.png)

Of course this is automated through a function :)

**Reply by PrzemyslawKlys** [reply]
I already have this fixed in another branch https://github.com/EvotecIT/PSWriteHTML/compare/master...LibsUpdate, basically being able to set Color on the  New-ChartBar

**Reply by amanfromfrance** [reply]
oh great!! Thanks for your help!
